src/daemon/rpmostreed-transaction.cxx

```diff
@@ -215,14 +218,30 @@ static void
 transaction_progress_changed_cb (OstreeAsyncProgress *progress,
                                  RPMOSTreeTransaction *transaction)
 {
+  RpmostreedTransaction *self = RPMOSTREED_TRANSACTION (transaction);
+  RpmostreedTransactionPrivate *priv = rpmostreed_transaction_get_private (self);
+  // Write progress messages to the journal at this periodicity
+  // so that admins can watch progress of pulls via `journalctl`.
+  constexpr uint32_t PROGRESS_JOURNAL_PERIODIC_SECS = 15;
+
   guint64 start_time = ostree_async_progress_get_uint64 (progress, "start-time");
   guint64 bytes_transferred = ostree_async_progress_get_uint64 (progress, "bytes-transferred");
   guint64 elapsed_secs = 0;
   guint64 bytes_sec = 0;
 
+  gint64 current_monotonic = g_get_monotonic_time ();
+  gint64 elapsed = MAX(current_monotonic, priv->last_progress_journal) - priv->last_progress_journal;
+  bool emit_journal = (elapsed / G_USEC_PER_SEC) > PROGRESS_JOURNAL_PERIODIC_SECS;
```

Comment:
Minor: could constify this.

Reply:
It's true of the `guint64 elapsed` too and all the other variables...I feel like this bit will solve itself as we oxidize.